src/colormaps.cpp

```diff
@@ -0,0 +1,181 @@
+#include "spark_dsg/colormaps.h"
+
+#include <algorithm>
+#include <random>
+
+namespace spark_dsg::colormaps {
+
+namespace {
```

Comment:
Is there a reason this is in anonymous namespace? Seems like standard helper functions to me.

Reply:
I'm trying to be a little better about avoiding odr issues (even if it's unlikely someone else is going to define a `spark_dsg::colormaps::exponentialOffsetId` function elsewhere in the code. Happy to add a declaration in the header and expose it publicly though if you think it's useful downstream independent of `rainbowId`